src/sql/workbench/services/notebook/common/sqlSessionManager.ts

```diff
@@ -127,7 +127,8 @@ export class SqlSession implements nb.ISession {
 	}
 
 	changeKernel(kernelInfo: nb.IKernelSpec): Thenable<nb.IKernel> {
-		return Promise.resolve(this.kernel);
+		this._kernel = this._instantiationService.createInstance(SqlKernel);
```

Comment:
Can you explain the impact of this? For example do we need to dispose the previous kernel at all? Doesn't that have to happen to clean things up? Similarly, if it's for switching sessions do we need to cache last used connection, or how can we guarantee we have the "right one"?